packages/deepgram/src/deepgram-transcription-model.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  TranscriptionModelV1,
+  TranscriptionModelV1CallWarning,
+} from '@ai-sdk/provider';
+import {
+  combineHeaders,
+  convertBase64ToUint8Array,
+  createJsonResponseHandler,
+  parseProviderOptions,
+  postToApi,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import { DeepgramConfig } from './deepgram-config';
+import { deepgramFailedResponseHandler } from './deepgram-error';
+import { DeepgramTranscriptionModelId } from './deepgram-transcription-settings';
+import { DeepgramTranscriptionAPITypes } from './deepgram-api-types';
+
+// https://developers.deepgram.com/docs/pre-recorded-audio#results
+const deepgramProviderOptionsSchema = z.object({
+  // Base parameters
+  language: z.string().nullish(),
```

Comment:
possible improvement: I believe if we add JSDoc here on the schema, it'll display on the provider options type inline in the IDE